keps/sig-api-machinery/20180415-crds-to-ga.md

```diff
@@ -162,10 +163,65 @@ TODO: complete this list
+- For clarity, we include thersolds with and without conversion webhooks.
+
+**Custom Resources per Definition, with Conversion webhook:**
+
+| Object size | Threshold scope=namespace | Threshold: scope=cluster |
+| --- | --- | --- |
+| 10kb | 2500 | 12500 |
+| 25kb | 1000 | 5000 |
+| 50kb | 500 | 2500 |
+
+**Custom Resources per Definition, without Conversion webhook:**
+
+| Object size | Threshold scope=namespace | Threshold: scope=cluster |
+| --- | --- | --- |
+| 10kb | 5000 | 25000 |
+| 25kb | 2000 | 10000 |
+| 50kb | 1000 | 5000 |
+
+_Note: For `scope: Namespaced` custom resource definitions, the scope=namespace
+threshold indicates how many custom resource objects may be in each namespace,
+and the scope=cluster threshold indicates how many custom resource objects may
+be in the cluster total. For `scope: Cluster` custom resource definitions, only
+the scope=cluster threshold applies._
+
+**Custom Resource Definitions:**
+
+| Threshold scope=namespace | Threshold: scope=cluster |
+| --- | --- |
+| n/a | 500 |
+
+_Note: The Custom Resource Definition threshold was selected not due to the
+above SLI/SLOs, but instead due to the latency OpenAPI publishing, which is a
+background process that occurs asychroniously each time a Custom Resource
+Definition is updated. For 500 Custom Resource Definitions it takes slightly
+over 35 seconds for a a definition change to be visible via the OpenAPI schema
+endpoint._
+
+_Note: Given that the performance and scalability of conversion webhooks are the
+responsibility of their author, CRD scale targets are defined for conversion
+webhook latency that includes serialization/deserialization cost, but does not
+include the webhook conversion operation cost (i.e. the cost of the custom
+authored conversion routines for a particular conversion webhook is not
+accounted for)._
+
+GA custom resource scale targets were selected based on an [analysis of our current scale limits](https://docs.google.com/document/d/1tEstPQvzGvaRnN-WwGUWx1H9xHPRCy_fFcGlgTkB3f8).
+
+We ran a month long survey of Custom Resource Definition scale needs across Kubernetes mailing lists, slack channels and social media.
+Of the custom resource definitions surveyed, 96% are currently within these scale thresholds, 91% are within these thresholds for their anticipated future growth, and survey data provides useful guidance for our post-GA scalability work. See [survey of real-world custom resource usage](https://docs.google.com/document/d/1MTd_gDlpgBaT5sAKM4j6tQVeCFIT9J44RHzt2yWOK_g) for details.
+
+As part of GA the threshold and SLO documentation will be updated to make this clear and to
+encourage CRD authors to provide concrete thresholds/SLOs for their custom
+resource kinds to their users that account for the per resource conversion cost
+of their conversion webhook and/or size of their custom resources.
 
 ## Graduation Criteria
```

Comment:
Could you also remove the umbrella issue and link to the Github project https://github.com/orgs/kubernetes/projects/28?